clustered/server/src/main/java/org/ehcache/clustered/server/store/ClusterTierActiveEntity.java

```diff
@@ -588,12 +600,11 @@ public void synchronizeKeyToPassive(PassiveSynchronizationChannel<EhcacheEntityM
         mappingsToSend.set(new HashMap<>());
         size.set(0);
       }
-    }
-
-    // Sync message tracker as well, to prevent duplicates received on the passive
-    // We do it once on the last concurrency key
-    if (concurrencyKey == lastConcurrencyKey) {
-      syncChannel.synchronizeToPassive(new EhcacheMessageTrackerMessage(messageHandler));
+      Map<Long, Map<Long, EhcacheEntityResponse>> clientSourceIdTrackingMap = messageHandler.getTrackedClients()
```

Comment:
You probably again push that in the constructor. It will remove the code duplication as well.

Reply:
Still have that conditional about creating the message.

Reply:
Hum... Maybe have a `shouldbeSent` method on the message? Or just refactor that part in a dedicated method